### DISCUSSION TITLE
Passing data for each button of a SplitButton

### DISCUSSION THREAD

**Discussion by tarakeshp** [primefaces/community#834] [PrimeVue] [answered]
For each row in Datatable, a general feature to add a action splitbutton. Currently there is no way to pass row-id or data object for each button of a splitbutton such the command callback function could take it as an argument. This will be a really good fix, and feature that helps a lot instead of work arounds.
Also conditionally each button in the splitbutton should be shown or hidden. For example
Actions
     View
     Edit
     Delete

For few users they can see only View and Delete buttons in the splitbutton by adding v-if
For other users it could be only View button. The following code snippets would help only as a work around but not a complete solution.

```
<Column  header="Action" >
      <template #body="{ data }">
          <SplitButton
              icon="pi pi-ellipsis-v"
              text
              :model="[
                  { label: 'Edit', icon: 'pi pi-fw pi-pencil', command: () => editAction(data) },
                  { label: 'Delete', icon: 'pi pi-fw pi-trash', command: () => deleteAction(data) }
              ]"
          ></SplitButton>

      </template>
  </Column>
```

**Comment by tugcekucukoglu** [comment, marked as answer]
Column and DataTable has its own API and `body` slot of the Column component defines these options: https://primevue.org/datatable/#api.column.slots.body. Can not reach directly the rows' id but Column node and field can be find.
